src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/sifter/SifterSettingsMessage.java

```diff
@@ -105,7 +105,9 @@ protected void onExecute(
             final PlayerEntity player = ctxIn.getSender();
             if (player == null) return;
 
-            InventoryUtils.reduceStackInItemHandler(new InvWrapper(player.inventory), mesh);
+            if(!player.isCreative()) {
+                InventoryUtils.reduceStackInItemHandler(new InvWrapper(player.inventory), mesh);
```

Comment:
This is correct, however i just noticed that we don't check that the player (if in survival) has the item in their inventory on server side. As such this is exploitable (free upgrades) by a client using a bad version of the mod.

Can you please add a check up the top of this method to return if the player doesn't have the required item in their inventory (or ignore if in creative)

Thanks